report/5.Conclusion/FutureWork.tex

```diff
@@ -1,2 +1,35 @@
 % !TeX root = ../main.tex
 \section{Future work}
+
+\subsection{User-tests}
+As per the conclusion of this report, the HCL language has not been usability-tested.
+The target-group of HCL consists of two sub-groups, novices with no programming experience, and novices with limited programming experience.
+
+To conduct the usability-test, the sub-group with no programming experience, would have to be taught simple programming constructs in HCL and general programming paradigms.
+The sub-group with limited programming experience, would be asked to write simple programs in a language they know, and then write the equivalent programs in HCL.
+
+The first sub-group would be used to determine whether HCL is intuitive.
+The second sub-group would be used to evaluate whether HCL simplifies the programming paradigms enough.
+
+\subsection{Improvements to the language}
+While HCL is a complete language, there are still improvements that could be made.
+
+\textbf{Working Arduino}\\
+While the project group wanted HCL to run on the Arduino platform, this feature was not completed in time for the project hand-in.
+The Arduino does not allow lambdas, or the import of the library that is used in the C++ portion of the code.
+A solution could be to convert all lambdas to full functions and reference these instead.
```

Comment:
This is not a solution. At least not how I understand this line. Lambdas have their own "context" or lambda environment. So regular functions are not an option.